src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp

```diff
@@ -1939,6 +2022,92 @@ void Hydro_HancockPredict( real fc[][NCOMP_LR], const real dt, const real dh,
+//                g_FC_Var           : Array storing the face-centered conserved variables
+//                                     --> Should contain NCOMP_TOTAL_PLUS_MAG variables
+//                MinDens/Pres/Eint  : Density, pressure, and internal energy floors
+//                FracPassive        : true --> convert passive scalars to mass fraction during data reconstruction
+//                NFrac              : Number of passive scalars for the option "FracPassive"
+//                FracIdx            : Target variable indices for the option "FracPassive"
+//                JeansMinPres       : Apply minimum pressure estimated from the Jeans length
+//                JeansMinPres_Coeff : Coefficient used by JeansMinPres = G*(Jeans_NCell*Jeans_dh)^2/(Gamma*pi);
+//                EoS                : EoS object
+//
+// Return      : g_PriVar[][ CUBE(FLU_NXT) ]
+//-------------------------------------------------------------------------------------------------------
+GPU_DEVICE
+void Hydro_ConFC2PriCC_MHM(      real g_PriVar[][ CUBE(FLU_NXT) ],
+                           const real g_FC_Var [][NCOMP_TOTAL_PLUS_MAG][ CUBE(N_FC_VAR) ],
+                           const real MinDens, const real MinPres, const real MinEint,
+                           const bool FracPassive, const int NFrac, const int FracIdx[],
+                           const bool JeansMinPres, const real JeansMinPres_Coeff,
+                           const EoS_t *EoS )
+{
+   CGPU_LOOP( idx_fc, CUBE(N_FC_VAR) )
+   {
+#     ifdef LR_EINT
+      real Eint;
+      real* const Eint_Ptr = &Eint;
+#     else
+      real* const Eint_Ptr = NULL;
+#     endif
+
+      real OneCell[NCOMP_TOTAL_PLUS_MAG], OutCell[NCOMP_TOTAL_PLUS_MAG];
+      real fc[6][NCOMP_LR];
+
+      for (int f=0; f<6; f++)
+      for (int v=0; v<NCOMP_TOTAL_PLUS_MAG; v++)
+         fc[f][v] = g_FC_Var[f][v][idx_fc];
+
+      for (int v=0; v<NCOMP_TOTAL; v++)
+      {
+         OneCell[v] = (real)0.0;
+         for (int f=0; f<6; f++)   OneCell[v] += fc[f][v];
+         OneCell[v] /= (real)6.;
+      } // for (int v=0; v<NCOMP_TOTAL; v++)
+
+      for (int d=0; d<3; d++)
+      {
+         const int faceL = 2*d;
+         const int faceR = faceL+1;
+         OneCell[MAG_OFFSET+d] = (real)0.5 * (fc[faceL][MAG_OFFSET+d] + fc[faceR][MAG_OFFSET+d]);
+      } // for (int d=0; d<3; d++)
+
+      Hydro_Con2Pri( OneCell, OutCell, MinPres, FracPassive, NFrac, FracIdx,
+                     JeansMinPres, JeansMinPres_Coeff, EoS->DensEint2Pres_FuncPtr, EoS->DensPres2Eint_FuncPtr,
+                     EoS->AuxArrayDevPtr_Flt, EoS->AuxArrayDevPtr_Int, EoS->Table, Eint_Ptr );
+
+      for (int v=0; v<NCOMP_TOTAL_PLUS_MAG; v++)   g_PriVar[v][idx_fc] = OutCell[v];
+
+//    store Eint in the last variable for LR_EINT
+#     ifdef LR_EINT
+      g_PriVar[NCOMP_TOTAL_PLUS_MAG][idx_fc] = Hydro_CheckMinEint( Eint, MinEint );
+#     endif
```

Comment:
Remove it. Just add a comment in the `Note` header that this function does NOT store Eint in the last variable for LR_EINT.